### DISCUSSION TITLE
Variable expansion in raw text/code

### DISCUSSION THREAD

**Discussion by JCGoran** [typst/typst#1891] [Q&A] [answered]
Suppose my document looks like this:

```plaintext
#let v1 = "something"
#let v2 = "something else"

We should use `#v1`, not `#v2`.
```

When I compile it, I get as output:

![image](https://github.com/typst/typst/assets/25671271/39a36bd8-9fd1-4c2a-9f7e-2f13b3bcccde)

I was expecting the variables to be expanded in the above. On the other hand, when I remove the backticks, i.e. use:

```plaintext
#let v1 = "something"
#let v2 = "something else"

We should use #v1, not #v2.
```

I instead get:

![image](https://github.com/typst/typst/assets/25671271/e4c6cab7-a914-4694-b6a2-972e10fd2902)

Is this a bug, or is there some other way I should expand variables/expressions when using raw text mode? If it's the latter, could you point me to the relevant section in the docs? Thanks!

**Comment by EpicEricEE** [comment, marked as answer]
I assume this is intended, because it is supposed to interpret the content as raw text. You can however use the [`#raw`](https://typst.app/docs/reference/text/raw/) function instead:

```
#let v1 = "something"
#let v2 = "something else"

We should use #raw(v1), not #raw(v2).
```

**Reply by LaurenzV** [reply]
Yeah, this is probably the way to go.
